src/Services/Device.vala

```diff
@@ -227,7 +227,27 @@ public class Power.Services.Device : Object {
     }
 
     public string get_symbolic_icon_name_for_battery () {
-        return get_icon_name_for_battery () + "-symbolic";
+        if (!is_a_battery) {
+            return "preferences-system-power-symbolic";
+        }
+
+        var icon_name = "battery";
+
+        if (percentage >= 20) {
```

Comment:
Could be >10 and then add a min range of 20 so we get that caution battery icon until we're under 10%